libyara/proc/linux.c

```diff
@@ -337,6 +337,13 @@ YR_API YR_MEMORY_BLOCK* yr_process_get_next_memory_block(
         } while (c >= 0 && c != '\n');
       }
 
+      // Remove '\n' at the end of the line
+      len = strlen(buffer);
```

Comment:
I think we can reuse the result returned by `strrchr` instead of calling `strlen`, which would iterate over the buffer again. If `buffer` contains the `\n` character `strrchr` already returns a pointer to it, if not, you don't need to remove it.

Reply:
I have re-edited and pushed it, please review it again, thank you very much~ XD